src/syntaxtree.c

```diff
@@ -279,6 +482,49 @@ static Obj SyntaxTreeCompilePragma(Obj result, Stat stat)
     return result;
 }
 
+static Expr SyntaxTreeCodeIf(Obj node)
+{
+    UInt1 tnum = GetTypeTNum(node);
+    Obj   branches = ElmRecST(tnum, node, "branches");
+    UInt  len = LEN_PLIST(branches);
```

Comment:
So @ChrisJefferson pointed out in another comment (now on "outdated" code, so not immediately visible), where I suggested to change a `LEN_LIST` to `LEN_PLIST`, that we could go the other way, and change all PLIST functions to generic ones, for safety. I am also fine with this -- the alternative being to add lost of "is this really a pist" checks everywhere. What I do not like is code which calls `LEN_LIST` and then immediately `ELM_PLIST`, as this code used to do, and as unfortunately still quite some other kernel code does. It combines the worst of both worlds: inferior performance plus a risk of crashes...

Anyway, performance shouldn't matter much here, so I'd be fine if we replaced `LEN_PLIST` with `LEN_LIST`, `ELM_PLIST` with `ELM_LIST` (resp. `ELM0_LIST` in the few cases where it is needed), `SET_ELM_PLIST` with `ASS_LIST`, etc.